cpp/ql/lib/semmle/code/cpp/models/implementations/NoExceptFunction.qll

```diff
@@ -0,0 +1,11 @@
+import semmle.code.cpp.models.interfaces.NonThrowing
+
+/**
+ * A function that is annotated with a `noexcept` specifier (or the equivalent
+ * `throw()` specifier) guaranteeing that the function can not throw exceptions.
+ *
+ * Note: The `throw` specifier was deprecated in C++11 and removed in C++17.
+ */
+class NoexceptFunction extends NonThrowingFunction {
```

Comment:
Since the `throw()` signifier is deprecated & removed in the C++ standard I decided to just name this class after the `noexcept` signifier. This seems simpler than having the name reflect both the signifiers.